.github/workflows/cd-analytics-infra.yml

```diff
@@ -42,7 +52,7 @@ jobs:
       fail-fast: false
       matrix:
         directory: ["database", "service"]
-        envs: ${{ github.event_name == 'release' && fromJSON('["prod"]') || fromJSON('["dev", "staging"]') }} # deploy prod on releases, otherwise deploy staging and dev
+        envs: ${{ inputs.environment || github.event_name == 'release' && fromJSON('["prod"]') || github.ref_name == 'main' && fromJSON('["dev", "staging"]') || fromJSON('["dev"]') }}
```

Comment:
Have you tested this? I don't know if it needs to be `inputs.environment == null && inputs.environment || ...` versus what you have here

Reply:
I haven't yet, I was largely stealing off your suggestion in the ticket 😁

Reply:
Hmmmmmmmm let's merge and pray, testing this sounds annoying.